csl-data.json

```diff
@@ -1,431 +1,457 @@
+                "locator": {
+                    "type": "string"
+                },
+                "medium": {
+                    "type": "string"
+                },
+                "note": {
+                    "type": "string"
+                },
+                "original-publisher": {
+                    "type": "string"
+                },
+                "original-publisher-place": {
+                    "type": "string"
+                },
+                "original-title": {
+                    "type": "string"
+                },
+                "page": {
+                    "type": "string"
+                },
+                "page-first": {
+                    "type": "string"
+                },
+                "PMID": {
+                    "type": "string"
+                },
+                "PMCID": {
+                    "type": "string"
+                },
+                "publisher": {
+                    "type": "string"
+                },
+                "publisher-place": {
+                    "type": "string"
+                },
+                "references": {
+                    "type": "string"
+                },
+                "reviewed-title": {
+                    "type": "string"
+                },
+                "scale": {
+                    "type": "string"
+                },
+                "section": {
+                    "type": "string"
+                },
+                "shortTitle": {
+                    "type": "string"
+                },
+                "source": {
+                    "type": "string"
+                },
+                "status": {
+                    "type": "string"
+                },
+                "title": {
+                    "type": "string"
+                },
+                "title-short": {
+                    "type": "string"
+                },
+                "URL": {
+                    "type": "string"
+                },
+                "version": {
+                    "type": "string"
+                },
+                "year-suffix": {
+                    "type": "string"
+                }
+            }
+        }
+    },
+    "type": "array",
+    "items": {
+        "type": "object",
+        "allOf": [
+            {
+                "$ref": "#/definitions/requiredFields"
             },
-            "URL": {
-                "type": "string" 
+            {
+                "$ref": "#/definitions/dateFields"
             },
-            "version": {
-                "type": "string" 
+            {
+                "$ref": "#/definitions/nameFields"
             },
-            "volume": {
-                "type": [
-                    "string",
-                    "number" 
-                ] 
+            {
+                "$ref": "#/definitions/numberFields"
             },
-            "year-suffix": {
-                "type": "string" 
-            } 
-        },
-        "additionalProperties" : false 
```

Comment:
Nope, it was not.

Although, just so you're aware, there are some large APIs that currently serve "CSL JSON" responses that are peppered with all kinds of additional properties, so not sure how that would possibly affect those API consumers/maintainers...

<details>
<summary><strong>Here's an example of a typical response from CrossRef</strong></summary>
<pre>
{
    "indexed": {
        "date-parts": [[2018, 12, 18]],
        "date-time": "2018-12-18T19:41:10Z",
        "timestamp": 1545162070737
    },
    "reference-count": 0,
    "publisher": "BMJ",
    "license": [
        {
            "URL": "http://www.bmj.com/company/legal-information/terms-conditions/legal-information/tdm-licencepolicy",
            "start": {
                "date-parts": [[2018, 12, 13]],
                "date-time": "2018-12-13T00:00:00Z",
                "timestamp": 1544659200000
            },
            "delay-in-days": 0,
            "content-version": "tdm"
        }
    ],
    "content-domain": { "domain": ["bmj.com"], "crossmark-restriction": true },
    "abstract": "<jats:title>Abstract</jats:title>\n          <jats:sec>\n            <jats:title>Objective</jats:title>\n            <jats:p>To determine if using a parachute prevents death or major traumatic injury when jumping from an aircraft.</jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Design</jats:title>\n            <jats:p>Randomized controlled trial.</jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Setting</jats:title>\n            <jats:p>Private or commercial aircraft between September 2017 and August 2018.</jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Participants</jats:title>\n            <jats:p>92 aircraft passengers aged 18 and over were screened for participation. 23 agreed to be enrolled and were randomized.</jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Intervention</jats:title>\n            <jats:p>Jumping from an aircraft (airplane or helicopter) with a parachute versus an empty backpack (unblinded).</jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Main outcome measures</jats:title>\n            <jats:p>Composite of death or major traumatic injury (defined by an Injury Severity Score over 15) upon impact with the ground measured immediately after landing.</jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Results</jats:title>\n            <jats:p>\n              Parachute use did not significantly reduce death or major injury (0% for parachute\n              <jats:italic>v</jats:italic>\n              0% for control; P&gt;0.9). This finding was consistent across multiple subgroups. Compared with individuals screened but not enrolled, participants included in the study were on aircraft at significantly lower altitude (mean of 0.6 m for participants\n              <jats:italic>v</jats:italic>\n              mean of 9146 m for non-participants; P&lt;0.001) and lower velocity (mean of 0 km/h\n              <jats:italic>v</jats:italic>\n              mean of 800 km/h; P&lt;0.001).\n            </jats:p>\n          </jats:sec>\n          <jats:sec>\n            <jats:title>Conclusions</jats:title>\n            <jats:p>Parachute use did not reduce death or major traumatic injury when jumping from aircraft in the first randomized evaluation of this intervention. However, the trial was only able to enroll participants on small stationary aircraft on the ground, suggesting cautious extrapolation to high altitude jumps. When beliefs regarding the effectiveness of an intervention exist in the community, randomized trials might selectively enroll individuals with a lower perceived likelihood of benefit, thus diminishing the applicability of the results to clinical practice.</jats:p>\n          </jats:sec>",
    "DOI": "10.1136/bmj.k5094",
    "type": "article-journal",
    "created": {
        "date-parts": [[2018, 12, 13]],
        "date-time": "2018-12-13T22:34:15Z",
        "timestamp": 1544740455000
    },
    "page": "k5094",
    "update-policy": "http://dx.doi.org/10.1136/crossmarkpolicy",
    "source": "Crossref",
    "is-referenced-by-count": 0,
    "title": "Parachute use to prevent death and major trauma when jumping from aircraft: randomized controlled trial",
    "prefix": "10.1136",
    "author": [
        {
            "given": "Robert W",
            "family": "Yeh",
            "sequence": "first",
            "affiliation": []
        },
        {
            "given": "Linda R",
            "family": "Valsdottir",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Michael W",
            "family": "Yeh",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Changyu",
            "family": "Shen",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Daniel B",
            "family": "Kramer",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Jordan B",
            "family": "Strom",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Eric A",
            "family": "Secemsky",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Joanne L",
            "family": "Healy",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Robert M",
            "family": "Domeier",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Dhruv S",
            "family": "Kazi",
            "sequence": "additional",
            "affiliation": []
        },
        {
            "given": "Brahmajee K",
            "family": "Nallamothu",
            "sequence": "additional",
            "affiliation": []
        }
    ],
    "member": "239",
    "published-online": { "date-parts": [[2018, 12, 13]] },
    "container-title": "BMJ",
    "original-title": [],
    "language": "en",
    "link": [
        {
            "URL": "http://data.bmj.org/tdm/10.1136/bmj.k5094",
            "content-type": "unspecified",
            "content-version": "vor",
            "intended-application": "text-mining"
        },
        {
            "URL": "https://syndication.highwire.org/content/doi/10.1136/bmj.k5094",
            "content-type": "unspecified",
            "content-version": "vor",
            "intended-application": "similarity-checking"
        }
    ],
    "deposited": {
        "date-parts": [[2018, 12, 18]],
        "date-time": "2018-12-18T19:05:10Z",
        "timestamp": 1545159910000
    },
    "score": 1.0,
    "subtitle": [],
    "short-title": [],
    "issued": { "date-parts": [[2018, 12, 13]] },
    "references-count": 0,
    "alternative-id": ["10.1136/bmj.k5094"],
    "URL": "http://dx.doi.org/10.1136/bmj.k5094",
    "relation": {},
    "ISSN": ["0959-8138", "1756-1833"],
    "subject": ["General Medicine"],
    "container-title-short": "BMJ"
}
</pre>
</details>

Reply:
> so not sure how that would possibly affect those API consumers/maintainers...

Not at all... they currently return CSL that does not abide by the JSON Schema. I.e. this is the status quo. For Manubot, we actually use the JSON schema to filter the CSL returned by Crossref and remove all the crap.

Reply:
If we continue to use `allOf` and use refs as I do in this PR, then there's no way currently possible to have `additionalProperties: false`. So pick your poison. I personally think it's fine, but if you feel strongly about it, we'd have to scrap this.

cc: @rmzelle

Reply:
Ping @rmzelle -- My comment above still holds relevance.